backend/app/Http/Controllers/MarketplaceController.php

```diff
@@ -54,6 +55,37 @@ public function updateCard($card_id)
         return $this->getCardDetail($card_id);
     }
 
+    public function buyCard($card_id)
+    {
+        //update card user's id with current id
+        $user = auth()->user();
+        $card = Card::find($card_id);
+        if ($card->isUserOwner($user)) {
```

Comment:
missing an `!` inside the condition!

Reply:
shouldn't it be if the user owns the card they can't buy it again? That is why I removed !